core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/ErrorListener.java

```diff
@@ -0,0 +1,17 @@
+package com.mongodb.stitch.core.services.mongodb.remote.sync;
+
+import org.bson.BsonValue;
+
+/**
+ * ErrorListener receives non-network related errors that occur.
+ */
+public interface ErrorListener {
+
+  /**
+   * Called when an error happens for the given document id.
+   *
+   * @param documentId the _id of the document related to the error.
+   * @param error the error.
+   */
+  void onError(final BsonValue documentId, final Error error);
```

Comment:
[minor] Let's use Exception. An Error is really for classes of errors that require the program to terminate.

Reply:
Throwable is also an option but starting at Exception lets us move up to Throwable in the future.